### DISCUSSION TITLE
Feature Request: Automation app (Tasker, Macrodroid etc.) integration

### DISCUSSION THREAD

**Discussion by fivetoten** [vmiklos/plees-tracker#237] [Q&A]
Hi!

Just wondering if it's possible to add a feature allowing third party apps to trigger tracker functions. Use cases could be, toggle the tracker by scanning a NFC tag without tapping the phone or even with screen off; auto trigger start/stop at user configured time; etc.

I notice two ways of implementing this, hope it's helpful.

1. broadcast intents

   - [Track Work Time](https://zephyrsoft.org/trackworktime), [codes](https://github.com/mathisdt/trackworktime/blob/master/app/src/main/java/org/zephyrsoft/trackworktime/Constants.java)

     > [...] TWT can be triggered from other apps and just do the book-keeping of your work time. In this case, you have to create broadcast intents with the action _org.zephyrsoft.trackworktime.ClockIn_ or _org.zephyrsoft.trackworktime.ClockOut_. [...]

   - [Pano Scrobbler for LastFM codes](https://github.com/kawaiiDango/pScrobbler/blob/master/app/src/main/java/com/arn/scrobble/NLService.kt#L945)

2. [Tasker plugin](https://tasker.joaoapps.com/plugins-intro.html)

   - [Loop Habit Tracker](https://github.com/iSoron/uhabits)

 Thanks for the app!

**Comment by vmiklos** [comment]
Sounds interesting, are you planning to look into this yourself? Nite that the widget provided by the app is a similar situation: it invokes the main activity with a spent parameter to immediately start or stop the sleep after launching the activity.

**Comment by fivetoten** [comment]
I'm afraid that I'm not able to help with the implementation, as I barely know anything about android coding. So sorry :frowning_face:

**Comment by vmiklos** [comment]
No problem, I was just trying my luck. Other issues seem to be more important for me, so this is up for grabs for now. I may get back to this in the future.
